### PR TITLE
Add name-canonical and text-canonical

### PR DESCRIPTION
**Pull request by fatihi**

### CODE REVIEW

File: FiveRingsDb.Tests/Utils/StringExtensionsTests.cs [hunk 1]
```diff
@@ -0,0 +1,46 @@
+﻿using FiveRingsDb.Utils;
+using FluentAssertions;
+using NUnit.Framework;
+
+namespace FiveRingsDb.Tests.Utils
+{
+    public class StringExtensionsTests
```
Comment (programming-wolf) [inline FiveRingsDb.Tests/Utils/StringExtensionsTests.cs]: Seems to be just for `ToCanonical()`, not StringExtensions in general. We should adjust this name :)

File: FiveRingsDb/Models/CardType.cs [hunk 2]
```diff
@@ -17,6 +17,6 @@ public enum Type
         [EnumMember(Value = "stronghold")]
         Stronghold,
         [EnumMember(Value = "role")]
-        Role
+        Role,
```
Comment (programming-wolf) [inline FiveRingsDb/Models/CardType.cs]: trailing comma

File: FiveRingsDb/Models/Clan.cs [hunk 3]
```diff
@@ -19,6 +19,6 @@ public enum Clan
         [EnumMember(Value = "Scorpion")]
         Scorpion,
         [EnumMember(Value = "Unicorn")]
-        Unicorn
+        Unicorn,
```
Comment (programming-wolf) [inline FiveRingsDb/Models/Clan.cs]: trailing comma

File: FiveRingsDb/Models/Element.cs [hunk 4]
```diff
@@ -15,6 +15,6 @@ public enum Element
         [EnumMember(Value = "water")]
         Water,
         [EnumMember(Value = "all")]
-        All
+        All,
```
Comment (programming-wolf) [inline FiveRingsDb/Models/Element.cs]: trailing comma

File: FiveRingsDb/Models/RoleRestriction.cs [hunk 5]
```diff
@@ -17,6 +17,6 @@ public enum RoleRestriction
         [EnumMember(Value = "void")]
         Void,
         [EnumMember(Value = "water")]
-        Water
+        Water,
```
Comment (programming-wolf) [inline FiveRingsDb/Models/RoleRestriction.cs]: trailing comma..

File: FiveRingsDb/Models/SetName.cs [hunk 6]
```diff
@@ -35,6 +35,6 @@ public enum SetName
         [EnumMember(Value="underhand-of-the-emperor")]
         UnderhandOfTheEmperor,
         [EnumMember(Value="children-of-the-empire")]
-        ChildrenOfTheEmpire
+        ChildrenOfTheEmpire,
```
Comment (programming-wolf) [inline FiveRingsDb/Models/SetName.cs]: trailing comma

File: FiveRingsDb/Models/Side.cs [hunk 7]
```diff
@@ -11,6 +11,6 @@ public enum Side
         [EnumMember(Value = "dynasty")]
         Dynasty,
         [EnumMember(Value = "role")]
-        Role
+        Role,
```
Comment (programming-wolf) [inline FiveRingsDb/Models/Side.cs]: trailing comma

File: FiveRingsDb/Models/Trait.cs [hunk 8]
```diff
@@ -195,6 +195,6 @@ public enum Trait
         [EnumMember(Value = "wily-trader")]
         WilyTrader,
         [EnumMember(Value = "yojimbo")]
-        Yojimbo
+        Yojimbo,
```
Comment (programming-wolf) [inline FiveRingsDb/Models/Trait.cs]: trailing comma

File: FiveRingsDb/Utils/StringExtensions.cs [hunk 9]
```diff
@@ -0,0 +1,70 @@
+﻿using System.Globalization;
+using System.Text;
+using System.Text.RegularExpressions;
+
+namespace FiveRingsDb.Utils
+{
+    public static class StringExtensions
```
Comment (programming-wolf) [inline FiveRingsDb/Utils/StringExtensions.cs]: I don't like such generic names. It only deals with the canonical format. So `StringToCanonicalExtension` would be a more precise name.

File: FiveRingsDb/Utils/StringExtensions.cs [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+        private static string Lowercase(string input)
+        {
+            #pragma warning disable CA1308 // Normalize strings to uppercase
+            return input.ToLowerInvariant();
+            #pragma warning restore CA1308 // Normalize strings to uppercase
+        }
+
+        private static string RemoveDiacritics(string input)
+        {
+            var strFormD = input.Normalize(NormalizationForm.FormD);
```
Comment (programming-wolf) [inline FiveRingsDb/Utils/StringExtensions.cs]: Can we agree to use more expressive names for variables? :) I have no idea what strFormD is supposed to be. And I think I do not need to know. However, `normalisedInput` is a lot easier to understand. And maybe we can add one line of comment that explains what `FormD` and `FormC` is.

File: FiveRingsDb.Tests/Utils/StringExtensionsTests.cs (hunk 1 above)
Comment (fatihi) [reply FiveRingsDb.Tests/Utils/StringExtensionsTests.cs]: Yeah but the Name of the class is StringExtensions. So I wrote the class StringExtensionsTests.
The name for the tests is ToCanonical_Should_XYZ. 

I always went with that naming scheme, should we change that?

Comment (programming-wolf) [reply FiveRingsDb.Tests/Utils/StringExtensionsTests.cs]: Hm. I also wrote something about that a bit below. Still not sure, to be honest, what the perfect name would be.

File: FiveRingsDb/Utils/StringExtensions.cs (hunk 9 above)
Comment (fatihi) [reply FiveRingsDb/Utils/StringExtensions.cs]: That means you would make a seperate class for each extension method. I can live with that as well 👍

File: FiveRingsDb/Utils/StringExtensions.cs (hunk 10 above)
Comment (fatihi) [reply FiveRingsDb/Utils/StringExtensions.cs]: Honestly.... I don't know myself. I just found that code somewhere on stackoverflow... 😅

File: FiveRingsDb/Utils/StringExtensions.cs (hunk 9 above)
Comment (programming-wolf) [reply FiveRingsDb/Utils/StringExtensions.cs]: I think it's cleaner and easier to maintain, don't you think? I'd prefer it that way 🙂

**Pull request merged**